x/mint/abci.go

```diff
@@ -0,0 +1,103 @@
+package mint
+
+import (
+	"context"
+	"time"
+
+	"github.com/babylonlabs-io/babylon/x/mint/keeper"
+	"github.com/babylonlabs-io/babylon/x/mint/types"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// BeginBlocker updates the inflation rate, annual provisions, and then mints
+// the block provision for the current block.
+func BeginBlocker(ctx context.Context, k keeper.Keeper) {
+	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyBeginBlocker)
+
+	maybeUpdateMinter(ctx, k)
```

Comment:
just to confirm most of this logic is copy paste from cosmos-sdk mint ?

Reply:
Not really, it's entirely from Celestia https://github.com/celestiaorg/celestia-app/blob/main/x/mint/abci.go